Fixes #1717 - ignore FutureWarnings which are raised by pandas

timkpaine opened the pull request:
<!--

Thank you for submitting a PR to astroid!

To ease our work reviewing your PR, do make sure to mark the complete the following boxes.

-->

Long description and repro in https://github.com/PyCQA/astroid/issues/1717

## Steps

- [x] For new features or bug fixes, add a ChangeLog entry describing what your PR does.
- [x] Write a good description on what the PR does.

## Description

## Type of Changes

<!-- Leave the corresponding lines for the applicable type of change: -->

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Related Issue

<!--
If this PR fixes a particular issue, use the following to automatically close that issue
once this PR gets merged:

Closes #XXX
-->
https://github.com/PyCQA/astroid/issues/1717
https://github.com/PyCQA/pylint/issues/7205

jacobtylerwalls reviewed (commented):
Thank you for submitting a patch!

I looked at the blame to find why we're even escalating warnings to errors in the first place. It looks like the author was somewhat uncertain and it escaped discussion during review, see #785. I would be in favor of just changing the warningsfilter to "ignore" instead of "error" and then removing the warning types from the `except`. Otherwise, we could be back here with `UserWarning` or anything else. What do you think?

Pierre-Sassoulas commented on astroid/raw_building.py:
```diff
@@ -390,7 +390,7 @@ def object_build(
                 with warnings.catch_warnings():
                     warnings.simplefilter("error")
                     member = getattr(obj, name)
-            except (AttributeError, DeprecationWarning):
+            except (AttributeError, DeprecationWarning, FutureWarning):
                 # damned ExtensionClass.Base, I know you're there !
                 attach_dummy_node(node, name)
                 continue
```
I agree with @jacobtylerwalls that we should (try to) change the filtering logic to not escalate to error (it's risky if we miss one escalation, on the top of my head we're missing UserWarning, PendingDeprecationWarning and SyntaxWarning at least).Maybe something like this ?
```python
with warnings.catch_warnings():
    warnings.simplefilter("ignore")
    member = getattr(obj, name)
```    

 I'm wondering what we do in  ``attach_dummy_node(node, name)`` that could necessitate to list everything. But I just checked and we indeed missed a lot of warning, it's actually surprising that this was not reported before : https://docs.python.org/3/library/warnings.html#warning-categories. Are those specific warnings never used in any library pylint astroid is parsing ?

Pierre-Sassoulas reviewed (commented):
Thank you for opening the issue and fixing the problem yourself, much appreciated !

timkpaine commented:
Another option is to just catch the base warning class, but I agree just ignoring it all is probably better and certainly cleaner.

Pierre-Sassoulas reviewed (approved):
👌

jacobtylerwalls reviewed (changes requested):
Glad to have the fix in. I checked the regression test, and it [passes](https://github.com/jacobtylerwalls/astroid/pull/72) without your changes. Maybe a test that just uses `unittest.mock` to emit a `FutureWarning` as a side effect would be simpler?

This is a minor detail, but if you do end up needing to create a test module, perhaps you can put it with the others under `tests/testdata/python3/data`.

timkpaine commented:
@jacobtylerwalls I accidentally broke the test by adding `Float64Index=""` to the fake module, if you remove that it should fail. I've fixed that test and moved the module inside the data folder
